### ISSUE TITLE
Different batch_size results in different evaluation loss.

### ISSUE THREAD

**Issue by iminfine** [karpathy/llm.c#710]
I have modified the code in train_gpt2.py to enable loading a pre-trained model and conveniently obtaining its evaluation loss by inserting a break statement at the beginning of the training loop:

```
 if args.init_from =='eval' :  
    break
```
after https://github.com/karpathy/llm.c/blob/72698a5ba32482789fb3b14fcff8b6b6d20931d9/train_gpt2.py#L762

I have unexpectedly observed that when the evaluation batch size is varied from the size used during training, the evaluation loss also varies. This observation is counterintuitive. The evaluation loss should remain consistent regardless of the evaluation batch size.  Does any one know why?
